iib/web/api_v1.py

```diff
@@ -104,6 +104,30 @@ def _get_safe_args(args, payload):
     return safe_args
 
 
+def _get_unique_bundles(bundles):
+    """
+    Return list with unique bundles.
+
+    :param list bundles: bundles given in payload from original request
+    :return: list of unique bundles preserving order (python 3.6+)
+    :rtype: list
+    """
+    if not bundles:
+        return bundles
```

Comment:
It means not to continue in this when bundles are empty or None. And they might be and it is not an error.
And why not to catch it before? Because I would take care of it on multiple places instead of one as it is right now.

Reply:
I understand that multiple occurrences but it doesn't look as a good example how to do it.

Reply:
Another point of view is this:
Function `_get_unique_bundles` should return unique list of bundles.
And this might be taken as processing empty list and returning unique list of bundles - empty list.

What would be your suggestion?

Reply:
I added the comment where it would fit better. Line no. 305

Reply:
You change line 307 like I suggested but you didn't removed this line

Reply:
Or maybe not. It looks good.